### PR TITLE
[go] Update B3 header environment variable documentation

### PR DESCRIPTION
**Pull request by nsrip-dd**
### What does this PR do?

Update the documentation for configuring B3 header injection and extraction for
the Go tracer.

### Motivation

The Go tracer has changed the environment variables for configuring B3
headers injection and extraction, including a new environment variable
to configure both. See https://github.com/DataDog/dd-trace-go/pull/1603.

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/tracing/trace_collection/custom_instrumentation/go.md [hunk 1]
```diff
@@ -206,10 +206,16 @@ configuring injection/extraction styles. Two styles are
...
 
 Configure extraction styles using the environment variable:
-`DD_PROPAGATION_STYLE_EXTRACT=Datadog,B3`
+`DD_TRACE_PROPAGATION_STYLE_EXTRACT=Datadog,B3`
+
+Configure both styles using the environment variable:
+`DD_TRACE_PROPAGATION_STYLE=Datadog,B3`
+
+Values from the respective injection and extraction style environment variables
+take precedence over `DD_TRACE_PROPAGATION_STYLE`.
```
Comment (katiehockman) [inline content/en/tracing/trace_collection/custom_instrumentation/go.md:218]:
```suggestion
Configure both styles using the environment variable:
`DD_TRACE_PROPAGATION_STYLE=Datadog,B3`.
Note that if either `DD_TRACE_PROPAGATION_STYLE_INJECT` or 
`DD_TRACE_PROPAGATION_STYLE_EXTRACT` are set, those will take precedence.
```

**Review by katiehockman — APPROVED** [review]

File: content/en/tracing/trace_collection/custom_instrumentation/go.md (hunk 1 above)
Comment (katiehockman) [reply content/en/tracing/trace_collection/custom_instrumentation/go.md:218]: Just a slight wording suggestion.

Comment (nsrip-dd) [reply content/en/tracing/trace_collection/custom_instrumentation/go.md]: Thanks!

**Comment by nsrip-dd** [comment]
dd-trace-go v1.46.0 has been released with these changes, so this is ready to be reviewed and merged.

File: content/en/tracing/trace_collection/custom_instrumentation/go.md [hunk 2]
```diff
@@ -206,10 +206,15 @@ configuring injection/extraction styles. Two styles are
...
+`DD_TRACE_PROPAGATION_STYLE_INJECT=Datadog,B3`
 
 Configure extraction styles using the environment variable:
-`DD_PROPAGATION_STYLE_EXTRACT=Datadog,B3`
+`DD_TRACE_PROPAGATION_STYLE_EXTRACT=Datadog,B3`
+
+Configure both styles using the environment variable:
+`DD_TRACE_PROPAGATION_STYLE=Datadog,B3`.
+Note that if either `DD_TRACE_PROPAGATION_STYLE_INJECT` or 
+`DD_TRACE_PROPAGATION_STYLE_EXTRACT` are set, those will take precedence.
```
Comment (cswatt) [inline content/en/tracing/trace_collection/custom_instrumentation/go.md:217]:
```suggestion
*Note*: `DD_TRACE_PROPAGATION_STYLE_INJECT` and `DD_TRACE_PROPAGATION_STYLE_EXTRACT`, if set, take precedence.
```

**Review by cswatt — APPROVED** [review]

**Pull request merged**